### PR TITLE
🎨 Replace usage of pkgerr with fmt.Errorf

### PR DESCRIPTION
**Pull request by grdryn**
JIRA: https://issues.redhat.com/browse/INTLY-4187

Go 1.13 has the ability to wrap/unwrap errors in the standard library now, and `github.com/pkg/errors` is no longer needed.

### CODE REVIEW

File: pkg/controller/installation/products/threescale/reconciler.go [hunk 1]
```diff
@@ -269,12 +268,12 @@ func (r *Reconciler) getOauthClientSecret(ctx context.Context, serverClient pkgc
...
 	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: oauthClientSecrets.Name, Namespace: r.ConfigManager.GetOperatorNamespace()}, oauthClientSecrets)
 	if err != nil {
-		return "", errors.Wrapf(err, "Could not find %s Secret", oauthClientSecrets.Name)
+		return "", fmt.Errorf("Could not find %s Secret: %w", oauthClientSecrets.Name, err)
 	}
 
 	clientSecretBytes, ok := oauthClientSecrets.Data[string(r.Config.GetProductName())]
 	if !ok {
-		return "", errors.Wrapf(err, "Could not find %s key in %s Secret", string(r.Config.GetProductName()), oauthClientSecrets.Name)
+		return "", fmt.Errorf("Could not find %s key in %s Secret: %w", string(r.Config.GetProductName()), oauthClientSecrets.Name, err)
```
Comment (matskiv) [inline pkg/controller/installation/products/threescale/reconciler.go]: @grdryn this is the same case you mentioned to me in PM, `err` can be nil...

File: pkg/controller/installation/products/rhssouser/reconciler.go [hunk 2]
```diff
@@ -262,12 +261,12 @@ func (r *Reconciler) setupOpenshiftIDP(ctx context.Context, inst *v1alpha1.Insta
...
 	err := serverClient.Get(ctx, pkgclient.ObjectKey{Name: oauthClientSecrets.Name, Namespace: r.ConfigManager.GetOperatorNamespace()}, oauthClientSecrets)
 	if err != nil {
-		return pkgerr.Wrapf(err, "Could not find %s Secret", oauthClientSecrets.Name)
+		return fmt.Errorf("Could not find %s Secret: %w", oauthClientSecrets.Name, err)
 	}
 
 	clientSecretBytes, ok := oauthClientSecrets.Data[string(r.Config.GetProductName())]
 	if !ok {
-		return pkgerr.Wrapf(err, "Could not find %s key in %s Secret", string(r.Config.GetProductName()), oauthClientSecrets.Name)
+		return fmt.Errorf("Could not find %s key in %s Secret: %w", string(r.Config.GetProductName()), oauthClientSecrets.Name, err)
```
Comment (matskiv) [inline pkg/controller/installation/products/rhssouser/reconciler.go]: @grdryn I think here is the same issue, `err` can be nil and even if it's not, it's still unrelated to this error message.

File: test/e2e/integreatly_test.go [hunk 3]
```diff
@@ -445,11 +442,11 @@ func checkIntegreatlyNamespaceLabels(t *testing.T, f *framework.Framework, names
 		namespace := &corev1.Namespace{}
 		err := f.Client.Get(goctx.TODO(), client.ObjectKey{Name: intlyNamespacePrefix + namespaceName}, namespace)
 		if err != nil {
-			return errors.Wrap(err, "Error getting namespace: "+namespaceName+" from cluster")
+			return fmt.Errorf("Error getting namespace: %v from cluster: %w", namespaceName, err)
 		}
 		value, ok := namespace.Labels[label]
 		if !ok || value != "true" {
-			return errors.Wrap(err, "Incorrect label on integreatly namespace: "+namespaceName+". Expected: "+label+". Got: "+label+"="+value)
+			return fmt.Errorf("Incorrect %v label on integreatly namespace: %v. Expected: true. Got: %v. %w", label, namespaceName, value, err)
```
Comment (matskiv) [inline test/e2e/integreatly_test.go]: @grdryn similiar issue here^, `err` object is irrelevant to this error statement

File: test/e2e/integreatly_test.go [hunk 4]
```diff
@@ -460,13 +457,13 @@ func checkOperatorVersions(t *testing.T, f *framework.Framework, namespace strin
...
 	if err != nil {
-		return errors.Wrap(err, "Error getting installation CR from cluster when checking operator versions")
+		return fmt.Errorf("Error getting installation CR from cluster when checking operator versions: %w", err)
 	}
 
 	for product, version := range operatorVersions {
 		clusterVersion := installation.Status.Stages[stage].Products[operator.ProductName(product)].OperatorVersion
 		if clusterVersion != operator.OperatorVersion(version) {
-			return errors.Wrap(err, fmt.Sprintf("Error with version of %s operator deployed on cluster. Expected %s. Got %s", product, version, clusterVersion))
+			return fmt.Errorf("Error with version of %s operator deployed on cluster. Expected %s. Got %s. %w", product, version, clusterVersion, err)
```
Comment (matskiv) [inline test/e2e/integreatly_test.go]: @grdryn This seems to be the same case as other comments - `err` is irrelevant here.

**Comment by grdryn** [comment]
Thanks for pointing out those other cases @matskiv. I hate the way `err` is always in scope after its first declaration, and the convention to always re-use the name `err`. If we weren't, then the compiler would catch such things I think!

File: test/e2e/integreatly_test.go [hunk 5]
```diff
@@ -478,13 +475,13 @@ func checkOperandVersions(t *testing.T, f *framework.Framework, namespace string
...
 	if err != nil {
-		return errors.Wrap(err, "Error getting installation CR from cluster when checking operand versions")
+		return fmt.Errorf("Error getting installation CR from cluster when checking operand versions: %w", err)
 	}
 
 	for product, version := range operandVersions {
 		clusterVersion := installation.Status.Stages[stage].Products[operator.ProductName(product)].Version
 		if clusterVersion != operator.ProductVersion(version) {
-			return errors.Wrap(err, fmt.Sprintf("Error with version of %s deployed on cluster. Expected %s. Got %s", product, version, clusterVersion))
+			return fmt.Errorf("Error with version of %s deployed on cluster. Expected %s. Got %s. %w", product, version, clusterVersion, err)
```
Comment (matskiv) [inline test/e2e/integreatly_test.go]: @grdryn This seems to be the same case as other comments - err is irrelevant here.

**Comment by matskiv** [comment]
I dropped a few comments about the issues I found, but everything else looks good. I'll approve once the comments are addressed. :)
Great job @grdryn

**Comment by grdryn** [comment]
@matskiv thanks for the review, I believe all are addressed now. It's a good job the original one caused a test failure so we knew what to look for! :) At the same time, changing to `fmt.Errorf` would have resulted in a new error anyway, even if it was trying to wrap an existing error (nil or not).

**Comment by grdryn** [comment]
@matskiv thanks for the review, I believe all are addressed now. It's a good job the original one caused a test failure so we knew what to look for! :) At the same time, changing to `fmt.Errorf` would have resulted in a new error anyway, even if it was trying to wrap an existing error (nil or not).

File: test/e2e/integreatly_test.go (hunk 4 above)
Comment (grdryn) [reply test/e2e/integreatly_test.go]: This one "broke" the e2e test -- it was silently succeeding up to now when it should have been erroring. I've added a new commit to fix that, not sure if there will be additional failures after that.

**Comment by grdryn** [comment]
The tests pass! :balloon: Could you please give this one last lgtm @matskiv ?

**Pull request merged**